templates/glossary.md

```diff
@@ -0,0 +1,272 @@
+
+Lean (sub-)[structures](#structure) will often correspondingly name one of their fields as such (`carrier`).
+
+### class
+
+More fully, a *typeclass* (or *type class*).
+
+A Lean [structure](#structure) whose [instances](#instance) can be retrieved via [typeclass inference](#typeclass-inference).
+
+Distinct from the use of *class* in object oriented languages -- the word usage in functional programming languages comes from the [typeclasses of Haskell](https://en.wikipedia.org/wiki/Type_class).
+
+### core Lean
+
+As differentiated from [mathlib](#mathlib) or other community-authored Lean code, core Lean (or the "core library") refers to the portions of Lean which ship with the distribution of Lean itself.
+
+Historically, even the mathlib project itself was a part of core Lean, and was split off into its own separately maintained project afterwards to facilitate development speed.
+
+Some fundamental [declarations](#declarations) remain part of core Lean even after the split.
+Occasionally additional lemmas and definitions are still removed or migrated from the core community Lean 3 repository and into mathlib, should they conflict with newly developed mathlib code.
+
+The current portions of core Lean 3 can be found in the [Lean 3 community repository](https://github.com/leanprover-community/lean/tree/master/library), and for Lean 4 [similarly](https://github.com/leanprover/lean4/tree/master/src).
+
+### declaration
+
+A single Lean runtime object within a Lean environment.
+
+Or, ambiguously, any of a number of Lean commands which may define or declare such objects.
+
+Examples of such commands are the `def`, `theorem`, `constant` or `example` commands (and in Lean 3, the `lemma` command), amongst others.
+
+Further detail can be found in the [Lean documentation](https://leanprover.github.io/lean4/doc/declarations.html#basic-declarations).
+
+### diamond
+
+There are often many ways to turn a given structure into another one.
+A *diamond* is a collection of such ways.
+Diamonds are abundant because of [hierarchies](#hierarchy).
+[Typeclass inference](#typeclass-inference) will unpredictably take one of the paths for a given diamond, so we want this path to not matter.
+This amounts to making the [Type-valued](#Prop-vs-Type) fields of the different inferable structures [defeq](#defeq).
+When this is the case, we have a *defeq diamond*.
+
+### `equiv`
+
+As distinct from mathematical equality, [`equiv`](mathlib_docs/data/equiv/basic.html) allows for defining an equivalence or congruence of types.
+One important thing to note is that an `equiv` [holds data instead of being merely a proof](#bundled-vs-unbundled).
+
+### goal
+
+Within the context of interactively proving theorems in Lean, each targeted statement whose proof is in-progress.
+
+Or more broadly, type theoretically, the process of exhibiting a term of a specific desired type.
```

Comment:
This doesn't really match -- "goal" isn't a process.

Reply:
Slightly reworded. It's still a bit clunky but was the best I had for the minute, let me know how it looks now to you.

Reply:
Looks good to me!